Sources/Core/ApiClient/ApiSession.swift

```diff
@@ -17,7 +17,11 @@ extension ApiSession {
     ApiSession { request in
       #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
       if #available(macOS 12, iOS 15, tvOS 15, watchOS 8, *) {
-        return try await session.data(for: request)
+        let (data, response) = try await session.data(for: request)
+        guard let httpResponse = response as? HTTPURLResponse else {
+          throw URLError(.unsupportedURL)
```

Comment:
This is where we turn the generic and not very useful `URLResponse` (e.g. could be FTP) into a more useful `HTTPURLResponse` (has a status code). I think it's reasonable to throw an error if `response as? HTTPURLResponse` fails i.e. it's not actually HTTP, because that shouldn't be possible.